client/src/components/TransactionModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { Component } from "react";
+import { Button, Modal, Spinner } from "react-bootstrap";
+
+export const TransactionModalState = Object.freeze({
+  HIDDEN: 0,
+  ERROR: 1,
+  AWAITING_CONFIRMATION: 2,
+  CONFIRMED: 3,
+});
+
+const spinnerStyle = { height: "128px", width: "128px" };
+
+interface IProps {
+  modalState?: any; 
+  transactionHash?: any; 
+  confirmedText?: string; 
+  onHide?: any;
+}
+
+
+const TransactionModal = ({modalState, transactionHash, confirmedText, onHide} : IProps)  =>  {
+  const show = () => {
+    return modalState !== TransactionModalState.HIDDEN;
+  };
+
+  const getEtherscanUrl = (transactionHash: any) => {
+    // TODO: Support non-kovan network chains.
+    return `https://kovan.etherscan.io/tx/${transactionHash}`;
+  };
+
+  const spinner =
+    modalState < TransactionModalState.CONFIRMED ? (
+      <Spinner animation="border" role="status" style={spinnerStyle}>
+        <span className="sr-only">Loading...</span>
+      </Spinner>
+    ) : null;
```

Comment:
nit: something like optional.empty()

Reply:
I actually don't think there is an empty option in TS AFAIK. There is, however, an undefined option: https://howtodoinjava.com/typescript/undefined-vs-null/. But it doesn't seem to have much of an impact,